src/mongo/MongoTripod.class.php

```diff
@@ -657,11 +633,12 @@ protected function getDataUpdater()
     {
         if(!isset($this->dataUpdater))
         {
+            $readPreference = $this->collection->getReadPreference();
             $opts = array(
                 'defaultContext'=>$this->defaultContext,
                 OP_ASYNC=>$this->async,
                 'stat'=>$this->stat,
-                'readPreference'=>$this->collection->getReadPreference(),
+                'readPreference'=>$readPreference['type'],
```

Comment:
This confused me when I first saw it. Cos I stopped and thought *hang on ReadPreference is an Array but we are only using a literal value*

I know that `collection->getReadPreference()` always return an Array, but all we are doing here is plucking off the `type` property which is that literal, and using that.   Perhaps a different question is  *are we painting ourselves into a corner by relying on just the type property as our Read Preference, instead of passing the entire array around incase we decide to make use of the tagsets in the future*

http://php.net/manual/en/mongocollection.getreadpreference.php

As long as we are consistent internally (and it looks like we are, we only pass around the literal in our own $opts from what ive seen).